src/wp-includes/class-wp-duotone.php

```diff
@@ -0,0 +1,1064 @@
+		$name = self::get_css_custom_property_name( $slug );
+		return "var($name)";
+	}
+
+	/**
+	 * Get the URL for a duotone filter.
+	 *
+	 * @param string $filter_id The ID of the filter.
+	 * @return string The URL for the duotone filter.
+	 */
+	private static function get_filter_url( $filter_id ) {
+		return "url(#$filter_id)";
+	}
+
+	/**
+	 * Gets the SVG for the duotone filter definition.
+	 *
+	 * @param string $filter_id The ID of the filter.
+	 * @param array  $colors    An array of color strings.
+	 * @return string An SVG with a duotone filter definition.
+	 */
+	private static function get_filter_svg( $filter_id, $colors ) {
+		$duotone_values = array(
+			'r' => array(),
+			'g' => array(),
+			'b' => array(),
+			'a' => array(),
+		);
+
+		foreach ( $colors as $color_str ) {
+			$color = self::colord_parse( $color_str );
+
+			if ( null === $color ) {
+				$error_message = sprintf(
+					/* translators: %s: duotone colors */
+					__( '"%s" in theme.json settings.color.duotone is not a hex or rgb string.' ),
+					$color_str
+				);
+				_doing_it_wrong( __METHOD__, $error_message, '6.3.0' );
+			} else {
+				$duotone_values['r'][] = $color['r'] / 255;
+				$duotone_values['g'][] = $color['g'] / 255;
+				$duotone_values['b'][] = $color['b'] / 255;
+				$duotone_values['a'][] = $color['a'];
+			}
+		}
+
+		ob_start();
+
+		?>
+
+		<svg
+			xmlns="http://www.w3.org/2000/svg"
+			viewBox="0 0 0 0"
+			width="0"
+			height="0"
+			focusable="false"
+			role="none"
+			style="visibility: hidden; position: absolute; left: -9999px; overflow: hidden;"
+		>
+			<defs>
+				<filter id="<?php echo esc_attr( $filter_id ); ?>">
+					<feColorMatrix
+						color-interpolation-filters="sRGB"
+						type="matrix"
+						values="
+							.299 .587 .114 0 0
+							.299 .587 .114 0 0
+							.299 .587 .114 0 0
+							.299 .587 .114 0 0
+						"
+					/>
+					<feComponentTransfer color-interpolation-filters="sRGB" >
+						<feFuncR type="table" tableValues="<?php echo esc_attr( implode( ' ', $duotone_values['r'] ) ); ?>" />
+						<feFuncG type="table" tableValues="<?php echo esc_attr( implode( ' ', $duotone_values['g'] ) ); ?>" />
+						<feFuncB type="table" tableValues="<?php echo esc_attr( implode( ' ', $duotone_values['b'] ) ); ?>" />
+						<feFuncA type="table" tableValues="<?php echo esc_attr( implode( ' ', $duotone_values['a'] ) ); ?>" />
+					</feComponentTransfer>
+					<feComposite in2="SourceGraphic" operator="in" />
+				</filter>
+			</defs>
+		</svg>
+
+		<?php
+
+		$svg = ob_get_clean();
+
+		if ( ! SCRIPT_DEBUG ) {
+			// Clean up the whitespace.
+			$svg = preg_replace( "/[\r\n\t ]+/", ' ', $svg );
+			$svg = str_replace( '> <', '><', $svg );
+			$svg = trim( $svg );
+		}
+
+		return $svg;
+	}
+
+	/**
+	 * Returns the prefixed id for the duotone filter for use as a CSS id.
+	 *
+	 * Exported for the deprecated function wp_get_duotone_filter_id().
+	 *
+	 * @since 6.3.0
+	 * @deprecated 6.3.0
```

Comment:
Was it really included AND deprecated in 6.3? 🤔

Reply:
@ajlende can you have a look?

Reply:
This is accurate. The comment explains why.

> Exported for the deprecated function wp_get_duotone_filter_id().

In the process of refactoring duotone into a class in this last release cycle, `wp_get_duotone_filter_id` was deprecated. Pieces of the implementation were moved into private methods, so in order to keep the deprecated functions available they were added as deprecated public methods for the class.

Reply:
Ok thanks! I understand the idea.

Reply:
Is it worth adding an `@internal` annotation to the docblocks for items introduced and deprecated in the same version like this?

That will prevent the docs from been added to developer.wordpress.org.